### DISCUSSION TITLE
Add Arabic language in stamp

### DISCUSSION THREAD

**Discussion by MhmodTayel** [MatthiasValvekens/pyHanko#205] [Q&A] [answered]
I'm trying to add Arabic language to the stamp but I can't. any help, please
![ds](https://user-images.githubusercontent.com/38634328/212070400-8c25585e-81b4-4dbc-b5c8-3477a6ad3d71.png)

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @MhmodTayel, are you using the default stamp style with Arabic? That won't work, because it relies on a non-embedded font that only supplies (a small superset of) the standard Latin character set.

You'll have to use an OpenType font that has glyphs for Arabic to render your signature appearance. Currently, it also needs to have glyphs for the Latin alphabet since pyHanko's signature appearance generation code doesn't support mixing fonts.

There's an example in the test suite on how you can instantiate & use an OpenType font to get Arabic text in a signature stamp: https://github.com/MatthiasValvekens/pyHanko/blob/aa75ef83111a3ae9ed97c3f8c196d5ad4662e6b5/pyhanko_tests/test_stamp.py#L101-L112

(Note that this requires pyHanko to be installed with the `[opentype]` optional dependency group; this is especially crucial for Arabic and other complex scripts, since pyHanko relies on HarfBuzz to perform glyph shaping. Run `pip install pyHanko[opentype]` if you're not sure.)

**Comment by MhmodTayel** [comment]
I tried this example and it works for Arabic language just fine but it removed the English letters

**Reply by MatthiasValvekens** [reply]
Yes, as I pointed out above, you'll need to find a font that has glyphs for both Latin and Arabic. Currently, pyHanko's appearance generation code does not (yet) support switching between different fonts, or picking fonts based on available glyphs. :)

**Comment by MhmodTayel** [comment]
![ds2](https://user-images.githubusercontent.com/38634328/212142778-38076e25-8141-4e59-8a8d-756e0036ad05.png)

**Comment by MhmodTayel** [comment]
I found a font with glyphs for both Latin and Arabic that works just fine. Thanks, a lot
